File: Getting-Practice-with-Python-and-Javascript/5-Object Oriented Programming/Javascript/Excercises/program-5.js
// 5. **Clase Empleado**: Crea una clase Empleado con atributos para el nombre, la ID del empleado, el salario base, y las horas extra trabajadas. Incluye métodos para calcular el salario con las horas extra, y mostrar la información del empleado.

class Empleado {
    constructor(nombre, idEmpleado, salarioBase, horasExtra) {
        this.nombre = nombre
        this.idEmpleado = idEmpleado
        this.salarioBase = salarioBase
        this.horasExtra = horasExtra
    }

    info() {
        let resultado = `Nombre: ${this.nombre} \n`
        resultado += `ID Empleado: ${this.idEmpleado} \n`
        resultado += `Salario Base: ${this.salarioBase} \n`
        resultado += `Horas Extras: ${this.horasExtra} \n`
        return resultado;
    }

    calcularSalario() {
        let salarioTotal = this.salarioBase;
        if (this.horasExtra > 20) {
            salarioTotal += this.horasExtra * 20
        }
        else if (this.horasExtra > 10) {
            salarioTotal += this.horasExtra * 15
        }
        else {
            salarioTotal += this.horasExtra * 10
        }
        return `Salario Total + horas extra: $${salarioTotal}`;
    }
}



let empleadoOne = new Empleado("Imanol", "001", 4800, 14)
console.log(empleadoOne.info())
console.log(empleadoOne.calcularSalario()) 

File: Getting-Practice-with-Python-and-Javascript/5-Object Oriented Programming/Python/Excercises/program-5.py
# 5. **Clase Empleado**: Crea una clase Empleado con atributos para el nombre, la ID del empleado, el salario base, y las horas extra trabajadas. Incluye métodos para calcular el salario con las horas extra, y mostrar la información del empleado.


class Empleado:
    def __init__(self, nombre, idEmpleado, salarioBase, horasExtras):
        self.nombre = nombre
        self.idEmpleado = idEmpleado
        self.salarioBase = salarioBase
        self.horasExtras = horasExtras

    # Metodo para mostrar la informacion del Empleado.
    def __str__(self):
        resultado = f'Nombre: {self.nombre} \n'
        resultado += f'ID Empleado: {self.idEmpleado} \n'
        resultado += f'Salario Base: ${self.salarioBase} \n'
        resultado += f'Horas Extras Trabajadas: {self.horasExtras}hs'
        return resultado 
    
    # Metodo para calcular el salario con las horas extras trabajadas.
    def calcular_salario(self):
        salarioTotal = self.salarioBase;
        if self.horasExtras > 20:
            salarioTotal += self.horasExtras * 20
        elif self.horasExtras > 10:
            salarioTotal += self.horasExtras * 15
        else:
            salarioTotal += self.horasExtras * 10
        return salarioTotal
    






empleado1 = Empleado('Imanol Aguer','01987',4800,14)

print(empleado1)
print(empleado1.calcular_salario())
